cub/cmake/CubInstallRules.cmake

```diff
@@ -16,7 +16,24 @@ install(DIRECTORY "${CUB_SOURCE_DIR}/cub/cmake/"
 )
 # Need to configure a file to store the infix specified in
 # CMAKE_INSTALL_INCLUDEDIR since it can be defined by the user
-set(install_location "${CMAKE_INSTALL_LIBDIR}/cmake/cub")
+set(_CCCL_RELATIVE_LIBDIR "${CMAKE_INSTALL_LIBDIR}")
+if(_CCCL_RELATIVE_LIBDIR MATCHES "^${CMAKE_INSTALL_PREFIX}")
```

Comment:
This doesn't handle the case where `CMAKE_INSTALL_LIBDIR` is already a relative path. This check fails and emits the error below.

Initializing `_CCCL_RELATIVE_LIBDIR` with `CMAKE_INSTALL_FULL_LIBDUR` instead should work AIUI.

Reply:
Updated to properly handle relative paths.

Reply:
We don't want to use `CMAKE_INSTALL_FULL_LIBDIR` as that breaks the usage of `cmake --install --prefix` since the path won't get re-rooted to the value provided by `--prefix`.